crates/ruff_python_parser/src/string_token_flags.rs

```diff
@@ -0,0 +1,259 @@
+use bitflags::bitflags;
+
+use ruff_text_size::{TextLen, TextSize};
+
+bitflags! {
+    /// The kind of quote used for a string
+    #[derive(Default, Debug, Copy, Clone, PartialEq, Eq, Hash)]
+    struct QuoteFlags: u8 {
+        /// The string uses double quotes (`"`).
+        /// If this flag is not set, the string uses single quotes (`'`).
+        const DOUBLE = 1 << 0;
+
+        /// The string is triple-quoted:
+        /// it begins and ends with three consecutive quote characters.
+        const TRIPLE_QUOTED = 1 << 1;
+    }
+}
+
+/// [String and Bytes literals]: https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
+/// [PEP 701]: https://peps.python.org/pep-0701/
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub(crate) enum StringPrefix {
+    /// The string has a `u` or `U` prefix.
+    /// While this prefix is a no-op at runtime,
+    /// strings with this prefix can have no other prefixes set.
+    U,
```

Comment:
I think I would prefer more explicit names than just the characters. Like `Unicode` or `Raw`